frontend/src/slices/auth/auth.slice.ts

```diff
@@ -44,61 +47,87 @@ const { reducer, actions, name } = createSlice({
     },
   },
   extraReducers(builder) {
-    builder.addCase(signUp.pending, (state) => {
-      state.dataStatus = DataStatus.PENDING;
-    });
-    builder.addCase(signUp.fulfilled, (state, action) => {
-      state.error = null;
-      state.user = action.payload;
-      state.dataStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(authorizeDriverSocket.pending, (state) => {
-      state.socketDriverAuthStatus = DataStatus.PENDING;
-    });
-    builder.addCase(authorizeDriverSocket.rejected, (state) => {
-      state.socketDriverAuthStatus = DataStatus.REJECTED;
-    });
-    builder.addCase(authorizeDriverSocket.fulfilled, (state) => {
-      state.socketDriverAuthStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(signUp.rejected, (state, { payload }) => {
-      state.dataStatus = DataStatus.REJECTED;
-      state.error = payload ?? null;
-    });
-    builder.addCase(signIn.pending, (state) => {
-      state.dataStatus = DataStatus.PENDING;
-    });
-    builder.addCase(signIn.fulfilled, (state, action) => {
-      state.error = null;
-      state.user = action.payload;
-      state.dataStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(signIn.rejected, (state, { payload }) => {
-      state.dataStatus = DataStatus.REJECTED;
-      state.error = payload ?? null;
-    });
-    builder.addCase(getCurrent.pending, (state) => {
-      state.dataStatus = DataStatus.PENDING;
-    });
-    builder.addCase(getCurrent.fulfilled, (state, action) => {
-      state.user = action.payload;
-      state.dataStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(getCurrent.rejected, (state) => {
-      state.dataStatus = DataStatus.REJECTED;
-    });
-    builder.addCase(logOut.pending, (state) => {
-      state.dataStatus = DataStatus.PENDING;
-    });
-    builder.addCase(logOut.fulfilled, (state) => {
-      state.user = initialState.user;
-      state.dataStatus = DataStatus.FULFILLED;
-      state.socketDriverAuthStatus = DataStatus.IDLE;
-    });
-    builder.addCase(logOut.rejected, (state, { payload }) => {
-      state.dataStatus = DataStatus.REJECTED;
-      state.error = payload ?? null;
-    });
+    builder
+      .addCase(getCurrent.fulfilled, (state, action) => {
+        state.user = action.payload;
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(getCurrent.rejected, (state) => {
+        state.dataStatus = DataStatus.REJECTED;
+      })
+      .addCase(signIn.fulfilled, (state, action) => {
+        state.error = null;
+        state.user = action.payload;
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(signUp.fulfilled, (state, action) => {
+        state.error = null;
+        state.user = action.payload;
+        state.dataStatus = DataStatus.FULFILLED;
+      })
+      .addCase(editBusiness.fulfilled, (state, action) => {
+        const { firstName, lastName, phone, email, business } = action.payload;
+
+        if (state.user) {
```

Comment:
is there any logic in it? I think edit business can be fullfilled only for authorized user, so user should be present in state always